src/lib.rs

```diff
@@ -378,9 +807,65 @@ impl PolarSensor {
 
 /// Private helper to find characteristics from a uuid
 async fn find_characteristic(device: &Peripheral, uuid: Uuid) -> PolarResult<Characteristic> {
-    device.characteristics()
+    device
+        .characteristics()
         .iter()
         .find(|c| c.uuid == uuid)
         .ok_or(Error::CharacteristicNotFound)
         .cloned()
-}
\ No newline at end of file
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    // for async testing
+    macro_rules! aw {
+        ($e:expr) => {
+            tokio_test::block_on($e)
+        };
+    }
+
+    #[test]
+    fn type_push() {
+        let mut polar = aw!(PolarSensor::new("9F8F7521".to_string())).unwrap();
```

Comment:
Can you replace this and the one below with something that's invalid? Seeing a valid ID here makes me think it needs to exist, but if it's just dummy data to get an instance then perhaps something like "0" would be fine.

Reply:
ok